File: sample/src/main/java/com/urbanairship/sample/preference/EnableAdvertisingIdPreference.java
/* Copyright Airship and Contributors */

package com.urbanairship.sample.preference;

import android.content.Context;
import android.util.AttributeSet;

import com.urbanairship.UAirship;
import com.urbanairship.aaid.AdvertisingIdTracker;
import com.urbanairship.preference.UACheckBoxPreference;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class EnableAdvertisingIdPreference extends UACheckBoxPreference {

    public EnableAdvertisingIdPreference(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    public EnableAdvertisingIdPreference(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public EnableAdvertisingIdPreference(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected boolean getInitialAirshipValue(@NonNull UAirship airship) {
        return AdvertisingIdTracker.shared().isEnabled();
    }

    @Override
    protected void onApplyAirshipPreference(@NonNull UAirship airship, boolean enabled) {
        AdvertisingIdTracker.shared().setEnabled(enabled);
    }

}

File: urbanairship-debug/src/main/java/com/urbanairship/debug/deviceinfo/preferences/EnableAdvertisingIdPreference.kt
/* Copyright Airship and Contributors */

package com.urbanairship.debug.deviceinfo.preferences

import android.content.Context
import android.util.AttributeSet
import com.urbanairship.UAirship
import com.urbanairship.aaid.AdvertisingIdTracker
import com.urbanairship.preference.UACheckBoxPreference

class EnableAdvertisingIdPreference : UACheckBoxPreference {

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {}

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {}

    override fun getInitialAirshipValue(airship: UAirship): Boolean {
        return AdvertisingIdTracker.shared().isEnabled
    }

    override fun onApplyAirshipPreference(airship: UAirship, enabled: Boolean) {
        AdvertisingIdTracker.shared().isEnabled = enabled
    }
}
